### DISCUSSION TITLE
node js docker project rabbit mq not connection error

### DISCUSSION THREAD

**Discussion by ghost** [rabbitmq/rabbitmq-server#10408] [Other] [answered]
### Describe the bug

 The error you see below is an error. I want to create a rabbitmq server in my node js docker project and try to start from there. but I'm getting an error. The cause of the -111 error is not clear. I couldn't find anything about this in my research.
 
 at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1595:16) {
socketio-stack-app  |   errno: -111,
socketio-stack-app  |   code: 'ECONNREFUSED',
socketio-stack-app  |   syscall: 'connect',
socketio-stack-app  |   address: '192.168.112.2',
socketio-stack-app  |   port: 5672
socketio-stack-app  | }

### Reproduction steps

I cannot make a rabbitmq connection with my node js docker project. I want to do this

### Expected behavior

I cannot make a rabbitmq connection with my node js docker project. I want to do this

### Additional context

When I try to make a connection in my node js project, I get an error.
My docker-compose.yml file is like this
"
   rabbitmq:
     #build: ./rabbitmq
     image: rabbitmq:3-management
     #image: rabbitmq:3-management-alpine
     #container_name: rabbitmq-server-app
     #hostname: "rabbitmq-server-app"
     ports:
     # AMQP protocol port
       - 5672:5672
     # HTTP management UI
       - 15672:15672
     expose:
       - '5672'
     environment:
       RABBITMQ_DEFAULT_USER: your_username
       RABBITMQ_DEFAULT_PASS: your_password
       #RABBITMQ_DEFAULT_VHOST: "/"
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:15672"]
       interval: 30s
       timeout: 10s
       retries: 5
     volumes:
          - ~/.docker-conf/rabbitmq/data/:/var/lib/rabbitmq/
          - ~/.docker-conf/rabbitmq/log/:/var/log/rabbitmq
     networks:
       - my-net

"

**Comment by michaelklishin** [comment, marked as answer]
`ECONNREFUSED` is very specific: the client fails to establish a TCP connection.

Start with [inspecting node logs](https://rabbitmq.com/logging.html#logged-events) (with Docker they are usually set to standard output), then following the [guide on troubleshooting network connectivity](https://rabbitmq.com/troubleshooting-networking.html).

And please do not use issues for questions if a repo has GitHub Discussions enabled. That's just rude.

**Comment by michaelklishin** [comment]
I'm afraid our team does not have the cycles to troubleshoot your Docker setup, nor do we have any relevant details. Please take it from here using the above guides and, if all fails, [taking and inspecting a traffic capture](https://rabbitmq.com/amqp-wireshark.html) to gather more information.
